src/Controls/tests/Core.UnitTests/VisualElementTests.cs

```diff
@@ -237,5 +242,83 @@ public async Task ShadowDoesNotLeak()
 
 			Assert.False(reference.IsAlive, "VisualElement should not be alive!");
 		}
+
+		[Fact]
+		public void HandlerDoesntPropagateWidthChangesDuringBatchUpdates()
+		{
+			bool mapperCalled = false;
+
+			var mapper = new PropertyMapper<IView, ViewHandler>(ViewHandler.ViewMapper)
+			{
+				[nameof(IView.Height)] = (_,_) => mapperCalled = true,
+				[nameof(IView.Width)] = (_,_) => mapperCalled = true,
+			};
+
+			var mauiApp1 = MauiApp.CreateBuilder()
+				.UseMauiApp<ApplicationStub>()
+				.ConfigureMauiHandlers(handlers => handlers.AddHandler<BasicVisualElement>((services) => new BasicVisualElementHandler(mapper)))
+				.Build();
+
+			var element = new BasicVisualElement();
+			var platformView = element.ToPlatform(new MauiContext(mauiApp1.Services));
+
+			mapperCalled = false;
+			element.Frame = new Rect(0,0,100,100);
+			Assert.False(mapperCalled);
+		}
+
+		[Fact]
+		public void HandlerDoesPropagateWidthChangesWhenUpdatedDuringSizedChanged()
+		{
+			bool mapperCalled = false;
+
+			var mapper = new PropertyMapper<IView, ViewHandler>(ViewHandler.ViewMapper)
+			{
+				[nameof(IView.Height)] = (_,_) => mapperCalled = true,
+				[nameof(IView.Width)] = (_,_) => mapperCalled = true,
+			};
+			
+			var mauiApp1 = MauiApp.CreateBuilder()
+				.UseMauiApp<ApplicationStub>()
+				.ConfigureMauiHandlers(handlers => handlers.AddHandler<BasicVisualElement>((services) => new BasicVisualElementHandler(mapper)))
+				.Build();
+
+			var element = new BasicVisualElement();
+			var platformView = element.ToPlatform(new MauiContext(mauiApp1.Services));
+
+			element.SizeChanged += (_,_) => element.HeightRequest = 100;
+			mapperCalled = false;
+			element.Frame = new Rect(0,0,100,100);
+
+			Assert.True(mapperCalled);
+		}
+
+		[Fact]
+		public void WidthAndHeightRequestPropagateToHandler()
+		{
+			bool mapperCalled = false;
+
+			var mapper = new PropertyMapper<IView, ViewHandler>(ViewHandler.ViewMapper)
+			{
+				[nameof(IView.Height)] = (_,_) => mapperCalled = true,
+				[nameof(IView.Width)] = (_,_) => mapperCalled = true,
+			};
+
+			var mauiApp1 = MauiApp.CreateBuilder()
+				.UseMauiApp<ApplicationStub>()
+				.ConfigureMauiHandlers(handlers => handlers.AddHandler<BasicVisualElement>((services) => new BasicVisualElementHandler(mapper)))
+				.Build();
+
+			var element = new BasicVisualElement();
+			var platformView = element.ToPlatform(new MauiContext(mauiApp1.Services));
+
+			mapperCalled = false;
+			element.WidthRequest = 99;
+			Assert.True(mapperCalled);
```

Comment:
Maybe this will hide a thing that happens. Your fix is to make the code NOT call the mapper, but also setting the WR will call the mapper. So checking for True is scary...

Should this be an int counter? Before your fix, there will be 2 calls - first the WR then the W - but after it should just be the WR?

I wonder if this test should use ints and check for `Assert.Equal(1, mapperCalled)`